test/login_page_test.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import '../lib/features/login/presentation/pages/login_options.dart';
+
+final Size iPhone8Plus = Size(414.0, 716.0); // Size for iPhone8Plus
+final TestWidgetsFlutterBinding binding =
+    TestWidgetsFlutterBinding.ensureInitialized();
+
+void main() {
+  testWidgets(
+      "test that the initial page properly presents", _testFirstPagePresents);
+}
+
+// MARK: - Tests
+
+// test that the first page properly presents
+Future<void> _testFirstPagePresents(WidgetTester tester) async {
+  await binding.setSurfaceSize(iPhone8Plus);
+  await tester.pumpWidget(_buildTestableWidget(LoginOptions()));
+  expect(find.text('logo/name'), findsOneWidget);
+  expect(find.text('Sign up for new account'), findsOneWidget);
+  expect(find.text('Log in with Google'), findsOneWidget);
+  expect(find.text('Log in with Facebook'), findsOneWidget);
+  expect(find.byType(FlatButton), findsNWidgets(4));
+  expect(find.byType(Image), findsNWidgets(2));
+}
+
+// MARK: - Private
+
+Widget _buildTestableWidget(Widget widget) {
+  return MaterialApp(home: widget);
+}
```

Comment:
Is this just a UI test for checking if the widgets exist, not necesarily if they are built correctly. I guess there is no way to check that without manual.

Reply:
This screen is not doing anything so far (buttons dont do anything) but later on they will and we can do assertions of that type on those like taps, swipes, etc. `flutter_test` works as an integration testing framework and it should suffice. But if we want to visually assert stuff we should look into a snapshot framework somewhere. :)